parfive/tests/test_downloader.py

```diff
@@ -347,6 +359,25 @@ def test_empty_retry():
     dl.retry(f)
 
 
+def test_done_callback_error(tmpdir, testserver):
+    tmpdir = str(tmpdir)
+
+    def done_callback(filepath, url, error):
+        if error is not None:
+            Path("callback.error").touch()
```

Comment:
I would use a variable for this, I think the following will work, though I haven't used `nonlocal` in a long time.

```suggestion
   
    at_least_one_error = False
    def done_callback(filepath, url, error):
        nonlocal at_least_one_error
        if error is not None:
            at_least_one_error = True
    assert at_least_one_error
```